### PR TITLE
Unit Testing

### PR DESCRIPTION
**Pull request by DarkxDev**
- Wrote unit tests using Rspec.
- Add unit tests for all the methods in all the classes: `Book`, `Person`, `Student`, `Teacher`, `Classroom`, and `Rental`.

### CODE REVIEW

File: rspec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  it 'creates a new rental object' do
+    rental = @person.add_rental(@book, '22/22/2222')
+    expect(rental).to be_instance_of(Rental)
+  end
+
+  it "adds the rental object to the person's rentals" do
+    rental = @person.add_rental(@book, '22/22/2222')
+    expect(@person.rentals).to include(rental)
+  end
+end
```
Comment (bobb-Rob) [inline rspec/person_spec.rb:24]: Well done 👏🏾 

- [ ] In this Person_spec.rb file, We expect you do unit for all the methods within the Person class. I see you have `can_use_services?`,  `correct_name`, and `add_rental` methods. Also, I suggest you do unit test for the constructor method as well. See sample below 👇🏾 

```ruby
context "When creating a new instance of the Person class" do
    it "The constructor method returns a Person object with provided age, name, and empty rentals" do
      age = 27
      name = "Matt Smith"
      person = Person.new(age, name, false)

      expect(person.age).to eql age
      expect(person.name).to eql name
      expect(person.rentals).to eql []
    end

    it "The constructor method sets default value for the optional name parameter if it is not provided" do
      person = Person.new(age)
      expect(person.name).to eql "Unknown"
    end
  end

    context "When adding a new rentals" do
    
        it "The add_rental method returns a rental and adds its to the book" do
          book = Book.new("Title", "Author")
          person = Person.new(24, "Bob")
    
          rental = person.add_rental(book, "2021-05-04")
    
          expect(person.rentals).to eql [rental]
        end
      end

      context "When checking permissions" do
        
        # *check theres some positive and negative case scenarios
        # *can_use_services? negative scenario
        it "The can_use_services? method returns false if under age and no parent permission" do
          person = Person.new(14, "Rose", false)
          expect(person.can_use_services?).to be false
        end
        
        # *can_use_services? positive scenario
        it "The can_use_services? method returns true if over age" do
          person = Person.new(19, "Sam", false)
          expect(person.can_use_services?).to be true
        end
        
        it "The can_use_services? method returns true if has parent permission" do
          person = Person.new(14, "Rose", true)
          expect(person.can_use_services?).to be true
        end
      end  
     
```

Looking at the test above, the "**unit test**" tests the class implementation and its methods. Kindly redo your unit tests. It must not be exactly like what I have done above. But it does test the Person class logically.

File: rspec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  it "includes the book's title in the JSON string" do
+    expect(JSON.parse(@book.to_json)['title']).to eq('Requiem')
+  end
+
+  it "includes the book's author in the JSON string" do
+    expect(JSON.parse(@book.to_json)['author']).to eq('Hitman')
+  end
+end
```
Comment (bobb-Rob) [inline rspec/book_spec.rb:26]: - [ ] Kindly redo your unit test to reflect a logical implementation of this Book class. 
- [ ] Please revisit all your other classes and use the person class unit test example I provided as a guide to implementing unit testing for all your classes implement and thier methods.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @DarkxDev , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Required Changes ♻️

- [ ] Please ensure your coding partner contributes code to this pull request.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED♻️ 

## Hi @DarkxDev 👋, 

Well done for implementing the required changes 👏 
Your project is now complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**